universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java

```diff
@@ -506,4 +506,68 @@ public void clearArray_ignoresScalar() {
     data.maybeClearArray(Path.create("applicant.cars"));
     assertThat(data.asJsonString()).isEqualTo(expected);
   }
+
```

Comment:
praise: nice tests! 

question (blocking): Can we add a test for a multi-part expression with `&&` or `||`? I want to understand if it is possible

Maybe something like:
```
{
  "applicant": {
    "favorite_color": "blue"
    "address": {
      "zip": "12345"
    }
  }
}

$.applicant[?(@.favorite_color != 'green' && @.address.zip in ['12345', '45678'])]
```

Reply:
Or something like this one from the examples, where we compare against another value in the json structure: `$..book[?(@.price <= $['expensive'])]`

Reply:
We plan to implement `AND` and `OR` in java rather than JsonPath, so skipped that. Added two test assertions for referencing other values in applicant data for the right-hand comparison value.